src/frontend/js/components/CourseGlimpse/index.tsx

```diff
@@ -16,71 +16,115 @@ const messages = defineMessages({
+  ) => {
+    const container = event.currentTarget.closest('.course-glimpse');
+    if (document.activeElement !== container?.querySelector('.course-glimpse__link')) {
+      setFocused(false);
+    }
+  };
+  const focusEvents = {
+    onMouseEnter: onEnter,
+    onFocus: onEnter,
+    onMouseLeave: onLeave,
+    onBlur: onLeave,
+  };
+  return (
+    <div className={`course-glimpse ${focused ? 'course-glimpse--focused' : ''}`}>
+      {/* the media link is only here for mouse users, so hide it for keyboard/screen reader users.
+      Keyboard/sr will focus the link on the title */}
+      <div aria-hidden="true" className="course-glimpse__media">
+        <a tabIndex={-1} href={course.absolute_url} {...focusEvents}>
+          {/* alt forced to empty string because it's a decorative image */}
+          {course.cover_image ? (
             <img
               alt=""
-              className="category-badge__icon"
-              sizes={course.icon.sizes}
-              src={course.icon.src}
-              srcSet={course.icon.srcset}
+              sizes={course.cover_image.sizes}
+              src={course.cover_image.src}
+              srcSet={course.cover_image.srcset}
             />
-            <span className="category-badge__title">{course.icon.title}</span>
-          </span>
+          ) : (
+            <div className="course-glimpse__media__empty">
+              <FormattedMessage {...messages.cover} />
+            </div>
+          )}
+        </a>
+      </div>
+      <div className="course-glimpse__content">
+        <div className="course-glimpse__wrapper">
+          <h3 className="course-glimpse__title" title={course.title}>
+            <a className="course-glimpse__link" href={course.absolute_url} {...focusEvents}>
+              {course.title}
+            </a>
+          </h3>
+          {course.organization_highlighted_cover_image ? (
+            <div className="course-glimpse__organization-logo">
+              {/* alt forced to empty string because the organization name is rendered after */}
+              <img
+                alt=""
+                sizes={course.organization_highlighted_cover_image.sizes}
+                src={course.organization_highlighted_cover_image.src}
+                srcSet={course.organization_highlighted_cover_image.srcset}
+              />
+            </div>
+          ) : null}
+          <div className="course-glimpse__metadata course-glimpse__metadata--organization">
+            <Icon name="icon-org" title={intl.formatMessage(messages.organizationIconAlt)} />
+            <span className="title" title={course.organization_highlighted}>
+              {course.organization_highlighted}
+            </span>
+          </div>
+          <div className="course-glimpse__metadata course-glimpse__metadata--code">
+            <Icon name="icon-barcode" title={intl.formatMessage(messages.codeIconAlt)} />
+            <span>{course.code || '-'}</span>
```

Comment:
Is it meaning full to add a title to the `Icon` here? Is not it better to set an aria-label on the span ? Like `Course code 0000` ?

Reply:
_Just as a side-note, setting an aria-label on a span usually does not do anything, screen readers don't announce it. We would need to change the role of the span to trigger screen readers to see the aria-label._

I'm sure I'm overthinking this but it's a great question in general: is it better to label an image, or add hidden text?

We can decide to have a decorative icon, and add the label in a text wrapped in the offscreen class.

In that case, the actual text announced by screen readers will be, as you said, "Course code 0000", in the same sentence as one block.

Or we can label the icon like in the PR.

In that case, the screen reader will announce "Course code, Image", then "0000". So, a little more verbose. But the bonus of setting the icon title is that mouse users can hover the icon in order to show its title.

Both techniques are totally valid. I usually prefer the latter, because it helps mouse users too. But I know this is a questionable bonus, one could say: if we feel the need to have a text explanation of an icon, the solution should be to change the icon, or remove it and replace it by a text for everyone. Even more so because this "tooltip" solution is not available for mobile phone users.

Here I find it great to have tooltips for the icons. It's rather difficult to have a 100% obvious icon for an "organization", and the bar code icon is pretty small, it's easy to not get what it is directly. So a little help is welcomed.

Or we could do kind of both: have an Icon that has the "Course code" title _but_ also marked aria-hidden ; and "Course code" as offscreen text in the span. This adds a tooltip for mouse users, that is not announced by screen readers, because the offscreen text is announced. But now we want to be sure we understand in the code the distinction between an icon with a screen reader label or a mouse-tooltip label...

Reply:
Many thanks for this explanation !

With that in mind, it seems that your solution is the best compromise to offer a good experience for screen readers and mouse users while keeping code simple. So I would say, let's keep it !